### PR TITLE
gs: add nightshade undub crc

### PR DESCRIPTION
**Pull request by swosho**
Snice the list already contains some CRCs of the modified executables, there's no reason for the undubbed versions not to be included as well when necessary.

### CODE REVIEW

**Comment by tadanokojin** [comment]
I don't accept CRCs for undubs or protos.

With regards to already including CRCs for modified versions, I I've gone ahead and just removed the ones I could spot here https://github.com/PCSX2/pcsx2/commit/a68d834f8f6c9dcf64d3cc2755c54752e740d011 so if there are any others let me know and I'll remove them as well. I'd rather remove them all than start accepting new ones.

**Comment by swosho** [comment]
> I don't accept CRCs for undubs or protos.

Can I inquire why? Aren't they completely harmless? I can't imagine the maintainers of any other emulator ever actively gimping the modded titles support without a clear reason.

I'm cool with helping to nuke the remaining unofficial CRCs for consistency's sake. I only want it to be more transparent that it's the unwillingness to accept new CRCs to the list being the reason why some fan translated games, undubs, etc., have some visual glitches present on the hardware renderer.

**Comment by RedDevilus** [comment]
Better to make patches (pnachs) than relying on CRC hacks. While visually the same it will seperate the hacks from the core and perhaps improve upon them by not being complacent for a quick dirty fix.

**Comment by swosho** [comment]
Exactly. Making the CRC hacks setup (maybe optionally) external would've been a step in the right direction.

**Comment by RedDevilus** [comment]
That's what the GameDatabase is for and the cheats folder. You might also want to read this: https://github.com/PCSX2/pcsx2/issues/3568

**Comment by swosho** [comment]
There's the probable eventual solutions, fair enough.
Still, speaking for the here and now I can't comprehend why unofficial CRCs shouldn't be listed.

**Comment by RedDevilus** [comment]
Example: A game only worked fine on CRC hack level, it was removed, suddenly they made a patch in a day. It's more a longterm vision than shortterm. Besides we don't support stuff like romhacks and if someone comes along and need support how are we supposed to help? I understand that as user you want to put as many CRC hacks. But it's a way to grow to be less hacky.

TL;DR
image

**Comment by swosho** [comment]
>I understand that as user you want to put as many CRC hacks. But it's a way to grow to be less hacky.

Except I'm not advocating for putting in more CRC hacks, I only want the existing hack to be whitelisted for use with a specific modded version of the game.
Besides, if you remove the hack – it's gone for every whitelisted variant of the same game, official or otherwise.

In case the actual reason for gatekeeping is this:

>Besides we don't support stuff like romhacks and if someone comes along and need support how are we supposed to help?

Then perhaps you're simply not obliged to help with the unofficial / tampered versions of the games??
Is it worth it to actively cripple the romhacks compatibility over that?

**Comment by F0bes** [comment]
> Then perhaps you're simply not obliged to help with the unofficial / tampered versions of the games??

Yes, it makes giving support infinitely easier and prevents people blaming the emulator for bad romhacks, because it's not supported.

> Is it worth it to actively cripple the romhacks compatibility over that?

It's either we cater to all non-official mods or none at all, and as shown by https://github.com/PCSX2/pcsx2/commit/a68d834f8f6c9dcf64d3cc2755c54752e740d011, it'll be the latter.

**Comment by swosho** [comment]
Sabotaging the compatibility over such petty arbitrary reasons is mad. Change a single byte in the executable and you're SOL just because the PCSX2 maintainers said so.

> It's either we cater to all non-official mods or none at all, and as shown by a68d834, it'll be the latter.

In striking contrast, the actual PS2 hardware "caters by default" to all non-official mods and runs them all the same as the non-modded counterparts of those games.

**Comment by tadanokojin** [comment]
> Sabotaging the compatibility over such petty arbitrary reasons is mad. 

Thinking adding a line that enables a hack that patches out rendering is an improvement to "compatibility" is mad. You're not improving the compatibility of anything. You're just making it look less broken to most people, it's the illusion of compatibility. You've fixed nothing. There's been zero improvement.

> Change a single byte in the executable and you're SOL just because the PCSX2 maintainers said so.

So change a few extra bytes and improve the "compatibility".

> In striking contrast, the actual PS2 hardware "caters by default" to all non-official mods and runs them all the same as the non-modded counterparts of those games.

In striking contrast the PS2 doesn't require nopping out draws to improve "compatibility" but you're welcome to do that and explain whoever consumes this undub on real hw your logic on this one.

If all you're going to do is make accusations of pettiness and gatekeeping instead of addressing the arguments being presented then I don't see any reason to continue this discussion further. It's clearly not going anywhere.
